### PR TITLE
Avoid possible flaky builds

### PR DESCRIPTION
**Pull request by volodk85**
As @ywangd pointed out [here](https://github.com/elastic/elasticsearch/pull/110172#discussion_r1658065797) there is a risk of CI flakiness. It hasn't happened yet, but I agree that it might happen if the tests got allocated to the same forked jvm and run in parallel.

### CODE REVIEW

**Comment by volodk85** [comment]
@elasticmachine update branch

**Review by breskeby — APPROVED** [review]
gradle changes lgtm

File: server/src/test/java/org/elasticsearch/index/IndexSettingsOverrideTests.java [hunk 1]
```diff
@@ -29,12 +24,13 @@ public static IndexMetadata newIndexMeta(String name, Settings indexSettings) {
...
     }
 
-    @BeforeClass
-    public static void setSystemProperty() {
-        System.setProperty(IndexSettings.RefreshIntervalValidator.STATELESS_ALLOW_INDEX_REFRESH_INTERVAL_OVERRIDE, "true");
-    }
-
     public void testStatelessMinRefreshIntervalOverride() {
+        assumeTrue(
+            "This test depends on system property",
```
Comment (ywangd) [inline server/src/test/java/org/elasticsearch/index/IndexSettingsOverrideTests.java:29]: Nit:
```suggestion
            "This test depends on system property configured in build.gradle",
```

**Review by ywangd — APPROVED** [review]
LGTM

Thanks for addressing my comment.

**Pull request merged**